sdk/azidentity/interactive_browser_credential.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/pkg/browser"
+)
+
+// InteractiveBrowserCredential enables authentication to Azure Active Directory using an interactive browser to log in.
+type InteractiveBrowserCredential struct {
+	client       *aadIdentityClient
+	tenantID     string // Gets the Azure Active Directory tenant (directory) ID of the service principal
+	clientID     string // Gets the client (application) ID of the service principal
+	clientSecret string // Gets the client secret that was generated for the App Registration used to authenticate the client.
```

Comment:
I'm wondering if it's better to have clientSecret in an options struct for this credential? 

On a related, but more general note, shouldn't all of the credentials have an options struct to help for future additions?

Reply:
If clientSecret is optional then yeah it should go in an options struct (probably fold TokenCredentialOptions into this same struct).

IIRC, we added TokenCredentialOptions as the escape-hatch for optional parameters.  However, if we think it makes more sense to have per-ctor options then can we do away with TokenCredentialOptions?  Having a bunch of nested optional param structs seems a bit overkill.

Reply:
Yes we had decided to have TokenCredentialOptions as the escape-hatch...but thinking about it more I dont think it's the best solution. If I were to add clientSecret to TokenCredentialOptions then I think it would be confusing when these same options are used for other credentials that require clientSecret. So I'm leaning towards each credential having its own option type.

Reply:
Yes I see your point.

I forgot how many options are in TokenCredentialOptions so perhaps it needs to remain, although it might be better to not anonymously embed it in other options structs (i.e. make it another nil-able field).

Reply:
Done